### PR TITLE
Feature/app

### PR DESCRIPTION
**Pull request by waaaaal**
Prueba

### CODE REVIEW

**Comment by waaaaal** [comment]
Todo subido.

**Comment by waaaaal** [comment]
Correcicón primer ejercicio react

File: public/index.html [hunk 1]
```diff
@@ -9,7 +9,6 @@
       name="description"
       content="Web site created using create-react-app"
     />
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (dani-cabal) [inline public/index.html:12]: Why is this line removed?

File: public/index.html [hunk 2]
```diff
@@ -27,7 +26,6 @@
     <title>React App</title>
   </head>
   <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
```
Comment (dani-cabal) [inline public/index.html:30]: Same here

File: src/App.tsx [hunk 3]
```diff
@@ -1,26 +1,16 @@
 import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
+import './index.css';
+import { Header } from './Greaters';
```
Comment (dani-cabal) [inline src/App.tsx]: What is 'Greaters'? Please use descriptive names

File: src/App.tsx [hunk 4]
```diff
@@ -1,26 +1,16 @@
...
-    </div>
+    <>
+    <Header />
+    <MainContent />
+    <Footer />
+    </>
   );
 }
-
+// Prueba para PullRequest...
```
Comment (dani-cabal) [inline src/App.tsx]: Remove unnecessary comments

File: src/App.tsx [hunk 5]
```diff
@@ -1,26 +1,16 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+    <Header />
+    <MainContent />
```
Comment (dani-cabal) [inline src/App.tsx]: 'MainContent' is not enough descriptive. What's 'MainContent'? A registration form? A list of users? A news page?

File: src/Footer.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+function Footer() {
+  return (
+    <footer className="footer">
+      <div className="container-wrapper">
+        <p>&copy; Dani Cabal</p>
```
Comment (dani-cabal) [inline src/Footer.jsx]: Your name is not 'Dani Cabal'

File: src/Greaters.jsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    return <h1>{props.title}</h1>
+}
+
+export function Header(){
+return   <header className="header">
+<div className="container-wrapper">
+  <h1><a href='./'>Lista de tareas</a></h1>
+</div>
+</header>
+
```
Comment (dani-cabal) [inline src/Greaters.jsx]: Format this block properly

File: src/H.jsx [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export function H(props){
+    return <h1></h1>
```
Comment (dani-cabal) [inline src/H.jsx]: What's an 'H'. Don't use this kind of names.

What's the purpose of this component?

File: src/MainContent.jsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import TaskForm from "./TaskForm";
+function MainContent() {
+  return (
+    <main>
```
Comment (dani-cabal) [inline src/MainContent.jsx]: `<main>` will be probably shared by other parts in the app, as well as Footer and Header. Don't make it part of a particular component.

What you call `MainContent` is going to change in the future

File: src/TaskForm.jsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import TaskList from "./TaskList";
+
+function TaskForm() {
+  const [inputText, setInputText] = useState("");
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+  const [elementos, setElementos] = useState([]);
```
Comment (dani-cabal) [inline src/TaskForm.jsx]: Be consistent. Name everything in english.

File: src/TaskForm.jsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+      setIsButtonDisabled(true);
+      return;
+    }
+    setIsButtonDisabled(false);
+  };
+
+  const handleButtonChange = () => {
+    setElementos([...elementos, inputText]);
+
+    setInputText(" ");
```
Comment (dani-cabal) [inline src/TaskForm.jsx]: Set text as an empty string. A blank space is not an empty string.

File: src/TaskForm.jsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  return (
+    <>
+      <form className="task-form">
```
Comment (dani-cabal) [inline src/TaskForm.jsx]: Extract form to another component

File: src/TaskList.jsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import React, { useState, useRef } from "react";
+function TaskList(props) {
+  const liRef = useRef(null);
```
Comment (dani-cabal) [inline src/TaskList.jsx]: What's `liRef`? Use clear and descriptive names.

File: src/TaskList.jsx [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+  const handleListChange = () => {
+    if (liRef.current) {
+      liRef.current.classList.toggle("setTachado"); // Alterna la clase "tachado"
```
Comment (dani-cabal) [inline src/TaskList.jsx]: Be consistent. Use names in english.

Remove unnecesarry comment. I know what a toggle does.

File: src/index.tsx [hunk 15]
```diff
@@ -2,18 +2,16 @@ import React from 'react';
...
 import App from './App';
-import reportWebVitals from './reportWebVitals';
-
 const root = ReactDOM.createRoot(
   document.getElementById('root') as HTMLElement
 );
 root.render(
   <React.StrictMode>
-    <App />
+
```
Comment (dani-cabal) [inline src/index.tsx]: Remove unnecessary blank line

File: src/index.css [hunk 16]
```diff
@@ -1,13 +1,120 @@
...
+  border-radius: 5px;
+  cursor: pointer;
+}
+
+.button:disabled {
+  opacity: .5;
+  cursor: default;
+}
+
+.tachado{
```
Comment (dani-cabal) [inline src/index.css]: Be consistent. Do all the naming in english. Besides, `tachado` is confusing while it has a `'text-decoration': none`

File: src/index.css [hunk 17]
```diff
@@ -1,13 +1,120 @@
...
+  text-decoration: none;
+}
+
+.setTachado{
```
Comment (dani-cabal) [inline src/index.css]: Same here

**Review by dani-cabal — CHANGES_REQUESTED** [review]
Besides all comments in files, Please organize better your code. Move components to a separate folder, as well as styles.

File: src/TaskList.jsx [hunk 18]
```diff
@@ -1,17 +1,17 @@
 import React, { useState, useRef } from "react";
 function TaskList(props) {
-  const liRef = useRef(null);
+  const liReference = useRef(null);
```
Comment (dani-cabal) [inline src/TaskList.jsx]: I still don't know what is this. If you mean a task in the list you can call it `task`, `taskInList` or whatever you want as long as it is meaningful. `liRef` or `liReference` means basically nothing.

On the other hand, I don't know why are you using `useRef` hook

File: src/TaskList.jsx [hunk 19]
```diff
@@ -1,17 +1,17 @@
 import React, { useState, useRef } from "react";
```
Comment (dani-cabal) [inline src/TaskList.jsx]: Remove unused imports

File: src/TaskList.jsx [hunk 20]
```diff
@@ -1,17 +1,17 @@
...
 function TaskList(props) {
```
Comment (dani-cabal) [inline src/TaskList.jsx]: Please destructure `props` (and type them). While doing this, we can see at first sight all the props and what type are they.

File: src/TaskList.jsx [hunk 21]
```diff
@@ -1,17 +1,17 @@
...
-    if (liRef.current) {
-      liRef.current.classList.toggle("setTachado"); // Alterna la clase "tachado"
+    if (liReference.current) {
+      liReference.current.classList.toggle("setLineThrough");
     }
   };
   return (
     <section className="tasks">
       <ul className="task-list">
         {props.datosParaLista.map((elemento, index) => (
```
Comment (dani-cabal) [inline src/TaskList.jsx]: Be consistent. Use all names in english.

File: src/TaskList.jsx [hunk 22]
```diff
@@ -1,17 +1,17 @@
...
-          <li onClick={handleListChange} key={index} ref={liRef}>
+          <li onClick={handleListChange} key={index} ref={liReference}>
             {elemento}
```
Comment (dani-cabal) [inline src/TaskList.jsx]: Names in english. 

What is `elemento`? Is it a car? Is it a cow?

File: src/index.css [hunk 23]
```diff
@@ -110,11 +110,11 @@ main {
   cursor: default;
 }
 
-.tachado{
+.NotLineThrough{
```
Comment (dani-cabal) [inline src/index.css]: Naming convention in css [kebab-case](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/)

On the other hand (maybe I missed that), but I don't remember the use of this class in any file. If you don't plan to use it, remove it.

File: src/index.css [hunk 24]
```diff
@@ -110,11 +110,11 @@ main {
...
   text-decoration: none;
 }
 
-.setTachado{
+.setLineThrough{
```
Comment (dani-cabal) [inline src/index.css]: Apart from kebab-case naming, this is not the best name for the class. What if tomorrow I tell you to mark 'done' tasks with a red color or a yellow background instead of a line through? Change its name to something like 'done', 'completed', 'marked' or whatever you want, but not attached to its properties.

**Review by dani-cabal — CHANGES_REQUESTED** [review]

File: src/TaskForm.jsx [hunk 25]
```diff
@@ -1,10 +1,8 @@
 import React, { useState } from "react";
```
Comment (dani-cabal) [inline src/TaskForm.jsx]: This is a jsx file. Convert it to tsx and type it properly

File: src/TaskList.tsx [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+
+interface TaskListProps {
+  tasks: Task[];
+  handleCompleted: (index:number) => void
+}
+
+function TaskList({ tasks, handleCompleted }: TaskListProps) {
+
+
+  console.log("Tareas en TaskList:", tasks);
```
Comment (dani-cabal) [inline src/TaskList.tsx]: Remove console calls when doing a pr

File: src/TaskList.tsx [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+
+  return (
+    <section className="tasks">
+      <ul className="task-list">
+        {tasks.map((item, index) => (
+        <li className={item.completed ? 'complete' : 'noCompleted'}onClick={() => handleCompleted(index)} >
```
Comment (dani-cabal) [inline src/TaskList.tsx]: Add a space before onClick.

We don't really need a `noCompleted` class with a `text-decoration: none`. If we remove that class, text won't have the line. It could be something like

```typescript
<li className={item.completed ? 'complete': '' ...>
```

File: src/TaskManager.tsx [hunk 28]
```diff
@@ -0,0 +1,37 @@
...
+
+  const handleCompleted = (index: number): void => {
+    const updatedTasks = [...tasks];
+
+    updatedTasks[index].completed = !updatedTasks[index].completed;
+
+    setTasks(updatedTasks);
+  }
+
+  function createTask(taskText: string) {
```
Comment (dani-cabal) [inline src/TaskManager.tsx]: Missing return type

File: src/TaskList.tsx [hunk 29]
```diff
@@ -0,0 +1,31 @@
...
+  text: string;
+  completed: boolean;
+}
+
+interface TaskListProps {
+  tasks: Task[];
+  handleCompleted: (index:number) => void
+}
+
+function TaskList({ tasks, handleCompleted }: TaskListProps) {
```
Comment (dani-cabal) [inline src/TaskList.tsx]: Missing return type

File: src/index.css [hunk 30]
```diff
@@ -95,6 +95,15 @@ main {
   text-decoration: line-through;
 }
 
+.complete{
+  text-decoration: line-through;
+
+}
+.noCompleted{
```
Comment (dani-cabal) [inline src/index.css]: As commented before, this class is not needed

**Review by dani-cabal — CHANGES_REQUESTED** [review]

File: src/TaskForm.tsx [hunk 31]
```diff
@@ -1,10 +1,16 @@
 import React, { useState } from "react";
 
-function TaskForm({ createTask }) {
+
+interface TaskFormProp{
+  createTask: (input: string) => void;
+
+}
+
+function TaskForm({ createTask }:TaskFormProp){
```
Comment (dani-cabal) [inline src/TaskForm.tsx]: missing return type in function

File: src/TaskForm.tsx [hunk 32]
```diff
@@ -1,10 +1,16 @@
...
   const [inputText, setInputText] = useState("");
   const [isButtonDisabled, setIsButtonDisabled] = useState(true);
 
-  const handleInputChange = (e) => {
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (dani-cabal) [inline src/TaskForm.tsx]: Missing return type in function

File: src/TaskForm.tsx [hunk 33]
```diff
@@ -17,7 +23,7 @@ function TaskForm({ createTask }) {
   const handleButtonChange = () => {
```
Comment (dani-cabal) [inline src/TaskForm.tsx]: Missing return type in function

**Review by dani-cabal — CHANGES_REQUESTED** [review]

**Review by dani-cabal — APPROVED** [review]
